src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+    ).tag(config=True)
+
+    psf_parameterization = traits.ComponentName(
+        PsfMakerBase,
+        default_value="Psf3dMaker",
+        help="The parameterization of the point spread function to be used.",
+    ).tag(config=True)
+
+    bkg_parameterization = traits.ComponentName(
+        BackgroundRateMakerBase,
+        default_value="BackgroundRate2dMaker",
+        help="The parameterization of the background rate to be used.",
+    ).tag(config=True)
+
+    energy_bias_res_parameterization = traits.ComponentName(
+        EnergyBiasResolutionMakerBase,
+        default_value="EnergyBiasResolution2dMaker",
+        help=(
+            "The parameterization of the bias and resolution benchmark "
+            "for the energy prediction."
+        ),
+    ).tag(config=True)
+
+    ang_res_parameterization = traits.ComponentName(
+        AngularResolutionMakerBase,
+        default_value="AngularResolution2dMaker",
+        help="The parameterization of the angular resolution benchmark.",
+    ).tag(config=True)
+
+    sens_parameterization = traits.ComponentName(
+        SensitivityMakerBase,
+        default_value="Sensitivity2dMaker",
+        help="The parameterization of the point source sensitivity benchmark.",
+    ).tag(config=True)
+
+    full_enclosure = Bool(
+        False,
+        help=(
+            "Compute a full enclosure IRF by not applying a theta cut and only use"
+            " the G/H separation cut."
+        ),
+    ).tag(config=True)
+
+    aliases = {
+        "cuts": "IrfTool.cuts_file",
+        "gamma-file": "IrfTool.gamma_file",
+        "proton-file": "IrfTool.proton_file",
+        "electron-file": "IrfTool.electron_file",
+        "output": "IrfTool.output_path",
+        "chunk_size": "IrfTool.chunk_size",
+    }
+
+    flags = {
+        **flag(
+            "do-background",
+            "IrfTool.do_background",
+            "Compute background rate.",
+            "Do not compute background rate.",
+        ),
+        **flag(
+            "do-benchmarks",
+            "IrfTool.do_benchmarks",
+            "Produce IRF related benchmarks.",
+            "Do not produce IRF related benchmarks.",
+        ),
+        **flag(
+            "full-enclosure",
+            "IrfTool.full_enclosure",
+            "Compute a full-enclosure IRF.",
+            "Compute a point-like IRF.",
+        ),
+    }
+
+    classes = (
+        [
+            EventsLoader,
+        ]
+        + classes_with_traits(BackgroundRateMakerBase)
+        + classes_with_traits(EffectiveAreaMakerBase)
+        + classes_with_traits(EnergyMigrationMakerBase)
+        + classes_with_traits(PsfMakerBase)
+        + classes_with_traits(AngularResolutionMakerBase)
+        + classes_with_traits(EnergyBiasResolutionMakerBase)
+        + classes_with_traits(SensitivityMakerBase)
+    )
+
+    def setup(self):
+        self.opt_result = OptimizationResultStore().read(self.cuts_file)
```

Comment:
This is a clear sign that the comment I made above is correct: `OptimizationResultStore` should not be a class.

Its `read` should either be a function or a method on the `OptimizationResult` itself.